src/main/java/com/mraof/minestuck/client/renderer/tileentity/TotemLatheRenderer.java

```diff
@@ -0,0 +1,92 @@
+import net.minecraft.util.math.vector.Vector3f;
+import net.minecraftforge.client.model.data.IModelData;
+import net.minecraftforge.client.model.data.ModelDataMap;
+import software.bernie.geckolib3.geo.render.built.GeoBone;
+import software.bernie.geckolib3.renderers.geo.GeoBlockRenderer;
+import software.bernie.geckolib3.util.RenderUtils;
+
+import javax.annotation.Nullable;
+
+public class TotemLatheRenderer extends GeoBlockRenderer<TotemLatheDowelTileEntity>
+{
+	private TotemLatheDowelTileEntity lathe;
+	private IRenderTypeBuffer renderTypeBuffer;
+	
+	public TotemLatheRenderer(TileEntityRendererDispatcher rendererDispatcherIn)
+	{
+		super(rendererDispatcherIn, new TotemLatheModel());
+	}
+	
+	@Override
+	public RenderType getRenderType(TotemLatheDowelTileEntity animatable, float partialTicks, MatrixStack stack, @Nullable IRenderTypeBuffer renderTypeBuffer, @Nullable IVertexBuilder vertexBuilder, int packedLightIn, ResourceLocation textureLocation)
+	{
+		return RenderType.entityCutoutNoCull(textureLocation);
+	}
+	
+	@Override
+	public void renderEarly(TotemLatheDowelTileEntity animatable, MatrixStack stackIn, float ticks, @Nullable IRenderTypeBuffer renderTypeBuffer, @Nullable IVertexBuilder vertexBuilder, int packedLightIn, int packedOverlayIn, float red, float green, float blue, float partialTicks)
+	{
+		super.renderEarly(animatable, stackIn, ticks, renderTypeBuffer, vertexBuilder, packedLightIn, packedOverlayIn, red, green, blue, partialTicks);
+		this.lathe = animatable;
+		this.renderTypeBuffer = renderTypeBuffer;
+	}
+	
+	@Override
+	public void renderRecursively(GeoBone bone, MatrixStack stack, IVertexBuilder bufferIn, int packedLightIn, int packedOverlayIn, float red, float green, float blue, float alpha)
+	{
+		if(bone.getName().equals("totem"))
+		{
+			ItemStack dowel = this.lathe.getStack();
+			if(dowel.isEmpty())
+			{
+				return; // render nothing
+			}
+			
+			BlockState cruxiteDowel = MSBlocks.CRUXITE_DOWEL.defaultBlockState();
+			if(AlchemyHelper.hasDecodedItem(dowel))
+			{
+				cruxiteDowel = cruxiteDowel.setValue(MSProperties.DOWEL_BLOCK, CruxiteDowelBlock.Type.TOTEM).getBlockState();
+			}
+			
+			// position adjustments
+			stack.pushPose();
+			RenderUtils.translate(bone, stack);
+			RenderUtils.moveToPivot(bone, stack);
+			stack.translate(0.25, -0.375, 0.03125);
+			stack.mulPose(Vector3f.ZP.rotationDegrees(90));
+			
+			// render the dowel with the blockRenderer
+			ClientWorld level = Minecraft.getInstance().level;
+			BlockRendererDispatcher blockRenderer = Minecraft.getInstance().getBlockRenderer();
+			BlockPos pos = lathe.getBlockPos();
+			IModelData modelData = new ModelDataMap.Builder().build();
+			blockRenderer.renderModel(cruxiteDowel, pos, level, stack, renderTypeBuffer.getBuffer(RenderTypeLookup.getRenderType(cruxiteDowel, false)), false, level.random, modelData);
+			
+			stack.popPose();
+			renderTypeBuffer.getBuffer(RenderType.entityCutoutNoCull(this.getTextureLocation(lathe)));
```

Comment:
Looks like you're getting a render buffer without using it. Why?

Reply:
its to reset the buffer back to its state after using it to render the dowel. maybe theres a better way to achieve that

Reply:
i added a comment